kong/tools/grpc.lua

```diff
@@ -47,42 +53,63 @@ local function set_hooks()
       return date(t.seconds):fmt("${iso}")
     end,
     function (t)
-      if type(t) ~= "string" then
-        error (string_format("expected time string, got (%s)%q", type(t), tostring(t)))
-      end
+    if type(t) ~= "string" then
+      error (string_format("expected time string, got (%s)%q", type(t), tostring(t)))
+    end
 
-      local ds = date(t) - epoch
-      return {
-        seconds = ds:spanseconds(),
-        nanos = ds:getticks() * 1000,
-      }
-    end)
+    local ds = date(t) - epoch
+    return {
+      seconds = ds:spanseconds(),
+      nanos = ds:getticks() * 1000,
+    }
+  end)
 end
 
---- loads a .proto file optionally applies a function on each defined method.
-function grpc.each_method(fname, f, recurse)
-  local dir = pl_path.splitpath(pl_path.abspath(fname))
-  local p = protoc.new()
-  p:addpath("/usr/include")
-  p:addpath("/usr/local/opt/protobuf/include/")
-  p:addpath("/usr/local/kong/lib/")
-  p:addpath("kong")
-  p:addpath("kong/include")
-  p:addpath("spec/fixtures/grpc")
-
-  p.include_imports = true
-  p:addpath(dir)
-  p:loadfile(fname)
-  set_hooks()
-  local parsed = p:parsefile(fname)
+function _M.new()
+  local protoc_instance = protoc.new()
+  protoc_instance:addpath("/usr/include")
+  protoc_instance:addpath("/usr/local/opt/protobuf/include/")
+  protoc_instance:addpath("/usr/local/kong/lib/")
+  protoc_instance:addpath("kong")
+  protoc_instance:addpath("kong/include")
+  protoc_instance:addpath("spec/fixtures/grpc")
+  protoc_instance.include_imports = true
+
+  return setmetatable({
+    protoc_instance = protoc_instance,
+  }, _M)
+end
 
-  if f then
+function _M:addpath(path)
+  local protoc_instance = self.protoc_instance
+  if type(path) == "table" then
+    for _, v in ipairs(path) do
+      protoc_instance:addpath(v)
+    end
+  else
+    protoc_instance:addpath(path)
+  end
+end
+
+function _M:name_search(name)
```

Comment:
`name_search` looks a little strange, could we rename it to `search_by_name` or `search`?